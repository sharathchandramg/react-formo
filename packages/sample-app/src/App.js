import React, { Component } from 'react';
import './App.css';
import Form from 'react-web-formo';

const data = require('./schema/22-checklist.json');

const testData = [
  {
    type: 'phone',
    name: 'phone_number',
    label: 'Phone Number',
    required: true,
    editable: true,
    hidden: false,
  },
];

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      fields: data,
      formData: {},
    };
  }

  render() {
    return (
      <div className="App">
        <div style={{ width: '85%' }}>
          <Form
            className="form"
            ref={c => (this.formGenerator = c)}
            fields={this.state.fields}
            formData={this.state.formData}
          />
        </div>
        <div className="buttons" style={{ justifyContent: 'center' }}>
          <button
            className="button subBtn"
            type="button"
            onClick={() => this.formGenerator.getValues()}
          >
            Submit
          </button>
          <button
            className="button clrBtn"
            type="reset"
            onClick={() => this.formGenerator.resetForm()}
          >
            Clear Form
          </button>
          <button
            className="button vldBtn"
            type="button"
            onClick={() => this.formGenerator.onValidateFields()}
          >
            Validate Form
          </button>
        </div>
      </div>
    );
  }
}

export default App;
