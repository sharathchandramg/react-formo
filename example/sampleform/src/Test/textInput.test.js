import React from "react";
import { shallow } from 'enzyme';
import App from "../App";

describe(App,()=>{
  it('Renders App', () => {
    const app = shallow(<App />);
    expect(App).toMatchSnapshot();
  });

  it('Renders text input area',()=>{
    const app= shallow(<App/>);
    
  })

  it('Renders text input area with email and phone',()=>{
    const app= shallow(<App/>);
    const expectedOutput='<div>'+'<label>'+'<div>'+'<p>'+'</p>'+'</div>'+'<input type:"text"/>'+'</label>'+'</div>'
    +'<div>'+'<label>'+'<div>'+'<p>'+'</p>'+'</div>'+'<input type:"text"/>'+'</label>'+'</div>'+
    '<div>'+'<label>'+'<div>'+'<p>'+'</p>'+'</div>'+'<input type:"phone"/>'+'</label>'+'</div>'+
    '<div>'+'<label>'+'<div>'+'<p>'+'</p>'+'</div>'+'<input type:"email"/>'+'</label>'+'</div>';

    const realOutput=app.find('div.form').html();
    expect(realOutput.indexOf(expectedOutput)>-1).toEqual(true);
  })
});