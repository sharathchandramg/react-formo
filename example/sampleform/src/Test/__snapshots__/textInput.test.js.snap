// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders App 1`] = `[Function]`;

exports[`App Renders Appp 1`] = `[Function]`;

exports[`App Renders Text input 1`] = `[Function]`;
